pkg/receiver/rawk8seventsreceiver/k8sconfig.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This is a copy of https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/k8sconfig/config.go
+// We can't import it normally yet, since it's an internal package. We can use a normal import if/when we move to upstream
+// or upstream makes this non-internal.
+
+package rawk8seventsreceiver
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+
+	quotaclientset "github.com/openshift/client-go/quota/clientset/versioned"
+	k8s "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// AuthType describes the type of authentication to use for the K8s API
+type AuthType string
+
+// TODO: Add option for TLS once
+// https://go.opentelemetry.io/collector/issues/933
+// is addressed.
+const (
+	// AuthTypeNone means no auth is required
+	AuthTypeNone AuthType = "none"
+	// AuthTypeServiceAccount means to use the built-in service account that
+	// K8s automatically provisions for each pod.
+	AuthTypeServiceAccount AuthType = "serviceAccount"
+	// AuthTypeKubeConfig uses local credentials like those used by kubectl.
+	AuthTypeKubeConfig AuthType = "kubeConfig"
+	// AuthTypeTLS indicates that client TLS auth is desired
+	AuthTypeTLS AuthType = "tls"
+)
+
+var authTypes = map[AuthType]bool{
+	AuthTypeNone:           true,
+	AuthTypeServiceAccount: true,
+	AuthTypeKubeConfig:     true,
+}
+
+// APIConfig contains options relevant to connecting to the K8s API
+type APIConfig struct {
+	// How to authenticate to the K8s API server.  This can be one of `none`
+	// (for no auth), `serviceAccount` (to use the standard service account
+	// token provided to the agent pod), or `kubeConfig` to use credentials
+	// from `~/.kube/config`.
+	AuthType AuthType `mapstructure:"auth_type"`
+}
+
+// Validate validates the K8s API config
+func (c APIConfig) Validate() error {
+	if !authTypes[c.AuthType] {
+		return fmt.Errorf("invalid authType for kubernetes: %v", c.AuthType)
+	}
+
+	return nil
+}
+
+// createRestConfig creates an Kubernetes API config from user configuration.
+func createRestConfig(apiConf APIConfig) (*rest.Config, error) {
```

Comment:
Could we add link to original code?

Reply:
There's a comment with a link above the package definition start.